### ISSUE TITLE
[Experiment] The "Send You Notifications" pop-up is automatically triggered after signing from the "Hop from phone to laptop and back" screen

### ISSUE THREAD

**Issue by SoftVision-CosminMuntean** [mozilla-mobile/firefox-ios#14554]
**Prerequisites**
- Have a Nimubs iOS Set to Default in Onboarding experiment launched.
- Have a browser profile enrolled in the treatment branch of the Onboarding experiment.
- Firefox is not set as default app.

**Steps to reproduce**
1. Open the Firefox app with the profile from the prerequisites.
2. Tap the "Not now" on the first screen.
3. Tap the "Sign in" button from the Hop from phone to laptop and back" screen and finish the flow.
4. Observe the behavior.

**Expected behavior**
- The "Notifications help you do more with Firefox" is displayed and the Send You Notifications" pop-up is not triggered.

**Actual behavior**
- The "Notifications help you do more with Firefox" is displayed and the "Send You Notifications" pop-up is automatically triggered. 

**Device & build information**
- Device
	- iPhone 13 mini (iOS 16.5)
	
- Build version:
	- Firefox for iOS Beta 114.0 (31168)
	
**Notes**
- However, I am not sure if this is the intended behavior or if this is an issue. However, my understanding is that the "Send You Notifications" pop-up should be displayed only after the "Continue" button is tapped. 
- This issue is also reproducible on the Control branch.
- Here is a screen recording of the issue: [Gdrive](https://drive.google.com/file/d/1WwLTiohJeyJJFRyk-tizG4NN-DM17n3c/view?usp=share_link).

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-6467)

**Comment by data-sync-user** [comment]
➤ Andrei Bodea commented:

Verified as fixed on v114 (31330) with iPhone 13 Pro (15.7.1).

The notification was triggered anymore after a login.
